src/server/scripts/Spells/spell_dh.cpp

```diff
@@ -195,6 +196,35 @@ class spell_dh_chaos_strike : public AuraScript
     }
 };
 
+// 213010 Charred Warblades
+class spell_dh_charred_warblades : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DH_CHARRED_WARBLADES_HEAL });
+    }
+
+    bool CheckProc(ProcEventInfo& eventInfo)
+    {
+        return eventInfo.GetDamageInfo() && eventInfo.GetDamageInfo()->GetSchoolMask() & SPELL_SCHOOL_MASK_FIRE;
+    }
+
+    void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+    {
+        uint32 heal = CalculatePct(eventInfo.GetDamageInfo()->GetDamage(), aurEff->GetAmount());
+        GetTarget()->CastSpell(GetTarget(), SPELL_DH_CHARRED_WARBLADES_HEAL,
+            CastSpellExtraArgs(TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR)
+            .SetTriggeringAura(aurEff)
+            .AddSpellBP0(heal));
+    }
+
+    void Register() override
+    {
+        DoCheckProc += AuraCheckProcFn(spell_dh_charred_warblades::CheckProc);
+        OnEffectProc += AuraEffectProcFn(spell_dh_charred_warblades::HandleProc, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
```

Comment:
given we have PERIODIC_DUMMY here i assume that the heal is "gathered" and on each periodic tick it is released, we had a similar mechanic in spell_warr_brutal_vitality (just that it was damage there)